### PR TITLE
Feature: Adds Tag and RewardsTag

### PR DESCRIPTION
**Pull request by PetterSW**
¡This PR is built on top of #148!

# Why?

New components in the design system (called Label). 

# What?

- Adds [Tag](https://www.figma.com/file/Nu1V7557V7KovgpZCNDObQ/sats-app-components-new-(wip)?type=design&node-id=2109%3A32165&mode=dev)
- Adds [RewardsTag](https://www.figma.com/file/Nu1V7557V7KovgpZCNDObQ/sats-app-components-new-(wip)?type=design&node-id=2137%3A5284&mode=dev)

# Version Change

Minor

# UI Changes

image
image

### CODE REVIEW

**Comment by fespinoza** [comment]
@PetterSW here you should have the base to be the #148 branch to be able to see only the changes this PR does

**Comment by fespinoza** [comment]
mmm... I don't agree with the renaming to `SATSLabelLegacy`. Maybe we need a different way to call components built specifically for UIKit. Not all UIKit code is "legacy", since we may need to use UIKit in some places of the app, even though we prefer SwiftUI.

Also I'm not convinced this component should be called `SATSLabel`, it's too similar `SwiftUI.Label` but they are quite different, this is more to "tag" or "label" content. In that sense, names in Figma may not fit in our iOS context, so we don't need to always match 100%.

`SATSRewardsLabel` to me it's not a component, but something you build from "SATSLabel" with a specific text and style. Then it can be achieved with factory methods instead of its own component.

I'm open for a chat to debate these points

File: Tests/SATSCoreTests/ColorsTests.swift [hunk 1]
```diff
@@ -114,7 +114,7 @@ class ColorsTests: XCTestCase {
             .rewardsBlue,
             .rewardsSilver,
             .rewardsGold,
-            .rewardsPlatinum
+            .rewardsPlatinum,
```
Comment (sindrenm) [inline Tests/SATSCoreTests/ColorsTests.swift:117]: :heart:

File: Sources/SATSCore/BasicComponents/RewardsTag/RewardsTag.swift [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        var background: Color {
+            switch self {
+            case .blue: return .rewardsBlue
+            case .silver: return .rewardsSilver
+            case .gold: return .rewardsGold
+            case .platinum: return .rewardsPlatinum
+            }
+        }
+    }
+}
```
Comment (sindrenm) [inline Sources/SATSCore/BasicComponents/RewardsTag/RewardsTag.swift:49]: Swap these around? :nerd_face:

Comment (sindrenm) [reply Sources/SATSCore/BasicComponents/RewardsTag/RewardsTag.swift:49]:
```suggestion
public extension RewardsTag {
    enum Level: String {
        case blue = "Blue"
        case silver = "Silver"
        case gold = "Gold"
        case platinum = "Platinum"

        var background: Color {
            switch self {
            case .blue: return .rewardsBlue
            case .silver: return .rewardsSilver
            case .gold: return .rewardsGold
            case .platinum: return .rewardsPlatinum
            }
        }
    }
}

struct RewardsTag_Previews: PreviewProvider {
    static var previews: some View {
        VStack(spacing: .spacingM) {
            RewardsTag(level: .blue)
            RewardsTag(level: .silver)
            RewardsTag(level: .gold)
            RewardsTag(level: .platinum)
        }
    }
}

```

**Review by sindrenm — APPROVED** [review]
LGTM!

File: Sources/SATSCore/BasicComponents/Tag/Tag.swift [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            .satsFont(.small)
+            .padding(.vertical, .spacingXXS)
+            .padding(.horizontal, .spacingS)
+            .background(viewData.style.background)
+            .cornerRadius(4)
+    }
+}
+
+
+
```
Comment (fespinoza) [inline Sources/SATSCore/BasicComponents/Tag/Tag.swift:45]:
```suggestion
}

```

I see SwiftLint was not catching the issues in here

File: DemoApp/DemoApp/Views/ContentView.swift [hunk 4]
```diff
@@ -20,6 +20,8 @@ struct ContentView: View {
                     NavigationLink("Button Styles", destination: SATSButtonSwftUIDemoView())
                     NavigationLink("Custom Async Image", destination: CustomAsyncDemoView())
                     NavigationLink("SATSCell", destination: SATSCellDemoView())
+                    NavigationLink("Rewards Tag", destination: RewardsTagDemoView())
+                    NavigationLink("Tag", destination: TagDemoView())
```
Comment (fespinoza) [inline DemoApp/DemoApp/Views/ContentView.swift:24]: I would not consider this basic components, just normal components

**Review by fespinoza — APPROVED** [review]
A couple minor details to fix, but it looks good

**Comment by PetterSW** [comment]
@fespinoza RewardsTag dosent need to be a own component, but it is in the design system and if we at a later time want to change it, its way easier

**Pull request merged**